datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -413,4 +456,39 @@ private String getDatastreamName() {
 
     return Collections.unmodifiableList(metrics);
   }
+
+  /**
+   * This class encapsulates the topic and partition information for a given event
+   */
+  static class TopicPartitionInfo {
```

Comment:
Why do we need this special class when kafka has TopicPartition class?

Reply:
yeah, was debating whether to add a dependency on kafka or not for this module so I implemented my own. But you're right, it's just silly code duplication. I'll fix this to use Kafka's TopicPartition.